src/app/connect/connect.service.ts

```diff
@@ -71,15 +70,18 @@ export class ConnectService {
             });
         }
 
-        return this._client.get(conn, "/api").toPromise()
-            .then(_ => {
+        return this._client.get(conn, "/api")
+            .map(_ => {
                 this.complete(conn);
-                return Promise.resolve(conn);
+                this.save(conn);
+                return conn;
             })
-            .catch(_ => {
-                this.gotoConnect(true);
-
-                return this.ping(conn, new Date().getTime() + ConnectService.PING_TIMEOUT, popup);
+            .catch((e, __) => {
```

Comment:
I am wondering if we can refactor (e, __)  ==> (err, caught) and reuse caught instead of creating a new Observable object. Can you check that?

Reply:
we dont' have to, single underscore would work. Changed